lib/index.js

```diff
@@ -314,12 +312,16 @@ export function buildJsx(tree, options) {
           const attribute = attributes[index]
 
           if (attribute.type === 'JSXSpreadAttribute') {
-            if (fields.length > 0) {
-              objects.push({type: 'ObjectExpression', properties: fields})
-              fields = []
+            const {argument} = attribute
+            if (argument.type === 'ObjectExpression') {
+              properties.push(...argument.properties)
```

Comment:
This flattens objects by one level. I added this, because it’s consistent with what we had. IMO there’s no need to flatten objects, it’s the job of a minifier.

Following the logic of keeping this: Why not flatten object spread recursively? Why not remove duplicate keys?

Reply:
minifiers might not be used. I’m not saying we need to generate tiny code. But it’s fine to generate good small code that doesn’t build tons of objects and runs fast.
I think Babel also combines objects too

Reply:
Both Babel and TypeScript do this for one level indeed. I suppose consistency with other tooling is the best argument for this, although personally I find it weird. Who authors code like that anyway?